src/game.cpp

```diff
@@ -11442,6 +11442,22 @@ void game::unload( item &it )
         return;
     }
 
+    // Handle magazines as a special case
+    if( target->is_magazine() ) {
+        // Remove all contained ammo consuming half as much time as required to load the magazine
+        target->contents.erase( std::remove_if( target->contents.begin(), target->contents.end(), [&]( item& e ) {
```

Comment:
Erase-Remove is cleaner than manual iteration.
Do you need manual iteration here for any reason?